examples/c/ex_col_store.c

```diff
@@ -0,0 +1,555 @@
+
+    /* Generating random data to populate the weather table */
+    generate_data(weather_data);
+
+    home = example_setup(argc, argv);
+
+    /* Establishing a connection */
+    error_check(wiredtiger_open(home, NULL, "create,statistics=(fast)", &conn));
+
+    /* Establishing a session */
+    error_check(conn->open_session(conn, NULL, NULL, &session));
+
+    /* Create a table with columns and colgroup */
+    error_check(session->create(session, "table:weathertable",
+      "key_format=r,value_format=" WT_UNCHECKED_STRING(
+        5sHBBHBBHH5s) ",columns=(id,day,hour,temp,"
+                      "humidity,pressure,wind,"
+                      "feels_like_temp,loc_lat,loc_long,country),colgroups=(day_time,temperature,"
+                      "humidity_pressure,"
+                      "wind,feels_like_temp,location)"));
+
+    /* Create the colgroups */
+    error_check(session->create(session, "colgroup:weathertable:day_time", "columns=(day,hour)"));
+    error_check(session->create(session, "colgroup:weathertable:temperature", "columns=(temp)"));
+    error_check(session->create(
+      session, "colgroup:weathertable:humidity_pressure", "columns=(humidity,pressure)"));
+    error_check(session->create(session, "colgroup:weathertable:wind", "columns=(wind)"));
+    error_check(session->create(
+      session, "colgroup:weathertable:feels_like_temp", "columns=(feels_like_temp)"));
+    error_check(session->create(
+      session, "colgroup:weathertable:location", "columns=(loc_lat,loc_long,country)"));
+
+    /* Open a cursor on the table to insert the data ---[[[[ INSERT ]]]]--- */
+    error_check(session->open_cursor(session, "table:weathertable", NULL, "append", &cursor));
+    w = weather_data;
+    for (int i = 0; i < N_DATA; i++) {
+        cursor->set_value(cursor, w->day, w->hour, w->temp, w->humidity, w->pressure, w->wind,
+          w->feels_like_temp, w->loc_lat, w->loc_long, w->country);
+        error_check(cursor->insert(cursor));
+        w++;
+    }
+    /* Close cursor */
+    error_check(cursor->close(cursor));
+
+    /* Prints all the data in the database */
+    print_all_columns(session);
+
+    /* Update the temperature from Celsius to Fahrenheit */
+    print_temp_column(session);
+    update_celsius_to_fahrenheit(session);
+    print_temp_column(session);
+
+    /* Create indexes for searching */
+    error_check(session->create(session, "index:weathertable:hour", "columns=(hour)"));
+    error_check(session->create(session, "index:weathertable:country", "columns=(country)"));
+
+    /* Calling the example operations */
+    if (find_min_temp(session, start, end, &min_temp_result)) {
+        printf("The minimum temperature between %" PRIu16 " and %" PRIu16 " is %d.\n", start, end,
+          min_temp_result);
+    } else {
+        printf("Invalid start and end time range, please try again.");
+    }
+
+    max_temp_result = 0;
+    if (find_max_temp(session, start, end, &max_temp_result)) {
```

Comment:
As mentioned earlier, this function should follow convention and return void or int (0 for success or failure code).